lib/src/multires_stochastic_texture_synthesis.rs

```diff
@@ -1191,6 +1201,213 @@ impl PrerenderedU8Function {
+            ChunkSearchInfo {
+                x: chunk_x,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: x as i32,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: x as i32,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x + 1,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: (chunk_x + 1) * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x - 1,
+                y: chunk_y + 1,
+                center: false,
+                closest_point_on_boundary_x: chunk_x * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: (chunk_y + 1) * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x + 1,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: (chunk_x + 1) * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+            ChunkSearchInfo {
+                x: chunk_x - 1,
+                y: chunk_y - 1,
+                center: false,
+                closest_point_on_boundary_x: chunk_x * self.chunk_size as i32 - self.offset_x,
+                closest_point_on_boundary_y: chunk_y * self.chunk_size as i32 - self.offset_y,
+            },
+        ];
+        // Note locking all of them at different times is fine as opposed to
+        // all at once but this behavior should be noted
+        let mut tmp_result: Vec<(i32, i32, i32)> = Vec::with_capacity(k * 9);
+        result.clear();
+        result.reserve(k);
+
+        // this isn't really the kth best distance but should be good enough for how simple it is
+        let mut kth_best_squared_distance = i32::max_value();
+        for place_to_look in places_to_look.iter() {
+            if place_to_look.x >= 0
+                && place_to_look.x < self.grid_width as i32
+                && place_to_look.y >= 0
+                && place_to_look.y < self.grid_height as i32
+            {
+                let is_center = place_to_look.center;
+
+                // a tiny optimization to help us throw far away neighbors
+                // saves us a decent amount of reads
+                if !is_center {
+                    let squared_distance_to_closest_possible_point_on_chunk = (x as i32
```

Comment:
Ditto, a bunch of potentially dangerous `i32` here.

Reply:
👍